### PR TITLE
Make notIn and In function consistent

### PR DESCRIPTION
**Pull request by VincentLanglet**
#Subject 

The mock of the QueryBuilder was kinda weird

No changelog for tests

### CODE REVIEW

File: tests/Filter/QueryBuilder.php [hunk 1]
```diff
@@ -115,13 +113,17 @@ public function isNull($queryPart)
...
-     * @param string $parameter
+     * @param string       $alias
+     * @param string|array $parameter
      *
      * @return string
      */
     public function notIn($alias, $parameter)
     {
+        if (\is_array($parameter)) {
+            return sprintf('%s IN ("%s")', $alias, implode(',', $parameter));
```
Comment (core23) [inline tests/Filter/QueryBuilder.php]: Sure this is `IN` and not `NOT IN`?

Comment (VincentLanglet) [reply tests/Filter/QueryBuilder.php]: Such a bad copy paste, thanks

**Comment by VincentLanglet** [comment]
This should be better @core23. :)

File: tests/Filter/QueryBuilder.php [hunk 2]
```diff
@@ -48,20 +48,18 @@ public function expr()
     }
 
     /**
-     * @param string $name
-     * @param string $value
+     * @param string       $alias
+     * @param string|array $parameter
```
Comment (core23) [inline tests/Filter/QueryBuilder.php]:
```suggestion
     * @param string|string[] $parameter
```

File: tests/Filter/QueryBuilder.php [hunk 3]
```diff
@@ -126,13 +124,17 @@ public function isNotNull(string $queryPart): string
     }
 
     /**
-     * @param string $alias
-     * @param string $parameter
+     * @param string       $alias
+     * @param string|array $parameter
```
Comment (core23) [inline tests/Filter/QueryBuilder.php]:
```suggestion
     * @param string|string[] $parameter
```

**Comment by greg0ire** [comment]
@franmomu I'm afraid your last change in the admin bundle is going to make build fails il DoctrineORMAdmin, MongoDBAdmin, PHPCRAdmin, etc.

**Comment by franmomu** [comment]
> @franmomu I'm afraid your last change in the admin bundle is going to make build fails il DoctrineORMAdmin, MongoDBAdmin, PHPCRAdmin, etc.

I don't know if maybe it's better to just check in the `AbstractSonataAdminExtension` if the parameter exists, the test fails because the SonataAdminExtension is not registered. Anyway I fixed here and in MongoDBAdmin looks like there is no test with its extension. I'm taking a look at the PHPCRAdmin.

**Comment by VincentLanglet** [comment]
@greg0ire My build is not failing anymore, do you want to review ? :)

File: tests/Filter/QueryBuilder.php [hunk 4]
```diff
@@ -126,13 +124,17 @@ public function isNotNull(string $queryPart): string
...
-     * @param string $parameter
+     * @param string          $alias
+     * @param string|string[] $parameter
      *
      * @return string
      */
     public function notIn($alias, $parameter)
     {
+        if (\is_array($parameter)) {
+            return sprintf('%s NOT IN ("%s")', $alias, implode(',', $parameter));
```
Comment (phansys) [inline tests/Filter/QueryBuilder.php]:
```suggestion
            return sprintf('%s NOT IN ("%s")', $alias, implode(', ', $parameter));
```

File: tests/Filter/QueryBuilder.php [hunk 5]
```diff
@@ -48,20 +48,18 @@ public function expr()
     }
 
     /**
-     * @param string $name
-     * @param string $value
+     * @param string          $alias
```
Comment (phansys) [inline tests/Filter/QueryBuilder.php]:
```suggestion
     * @param string $alias
```

File: tests/Filter/QueryBuilder.php [hunk 6]
```diff
@@ -126,13 +124,17 @@ public function isNotNull(string $queryPart): string
     }
 
     /**
-     * @param string $alias
-     * @param string $parameter
+     * @param string          $alias
```
Comment (phansys) [inline tests/Filter/QueryBuilder.php]:
```suggestion
     * @param string $alias
```

File: tests/Filter/QueryBuilder.php [hunk 7]
```diff
@@ -48,20 +48,18 @@ public function expr()
...
      */
-    public function in($name, $value)
+    public function in($alias, $parameter)
     {
-        $this->query[] = 'in_'.$name;
-
-        if (\is_array($value)) {
-            return sprintf('%s IN ("%s")', $name, implode(',', $value));
+        if (\is_array($parameter)) {
+            return sprintf('%s IN ("%s")', $alias, implode(',', $parameter));
```
Comment (phansys) [inline tests/Filter/QueryBuilder.php]:
```suggestion
            return sprintf('%s IN ("%s")', $alias, implode(', ', $parameter));
```

File: tests/Filter/QueryBuilder.php (hunk 5 above)
Comment (VincentLanglet) [reply tests/Filter/QueryBuilder.php]: Why ? There is no lint error, and I saw a lot of function which align the params.
image

Comment (phansys) [reply tests/Filter/QueryBuilder.php]: I know, we haven't configured the CS rule `phpdoc_align`. My suggestion is just based on my own experience dealing with conflicts in other lines when some argument is renamed, added, removed or replaced.
Just ignore the suggestion if you don't agree :+1:

**Pull request merged**